### PR TITLE
Don't pass reduction variables known in an outer parfor to inner parfors when analyzing reductions.

### LINKED ISSUE #6414: After updating the Numba from 0.43 to 0.51.2, the parallel code is displaying the error
I have recently updated Numba in Anaconda at my machine from 0.43 to 0.51.2. The code which was running successfully with parallel=True is not running with prange. When I have replaced prange with range, the code executes but sequentially. Also, the sequential execution is taking much greater time with parallel=True in comparison to parallel=False.

The error message that I am getting in correspondence to my my program using parallel=True and prange is
```
TypingError: Internal error at <numba.core.typeinfer.CallConstraint object at 0x000001F463923C18>.
Failed in nopython mode pipeline (step: convert to parfors)
maximum recursion depth exceeded while calling a Python object
During: resolving callee type: type(CPUDispatcher(<function case1_flagi1_j1 at 0x000001F462EBC620>))

Enable logging at debug level for details.
```

### PR DESCRIPTION
**Pull request by DrTodd13**
Resolves #6414

### CODE REVIEW

File: numba/tests/test_parfors.py [hunk 1]
```diff
@@ -2343,6 +2343,33 @@ def test_impl(n):
...
+    def test_prange_nested_reduction1(self):
+        # issue6414
+        def test_impl():
+            A = 0
+            for k in range(1):
+                for i in range(1):
+                    if i == 0:
+                        A += 1
+            return A
+        self.prange_tester(test_impl)
```
Comment (stuartarchibald) [inline numba/tests/test_parfors.py:2371]: Does this reproduce this issue? I've tried:
```python
from numba import njit, prange
import numpy as np

@njit(parallel=True)
def test_impl():
    A = 0
    for k in prange(1):
        for i in prange(1):
            if i == 0:
                A += 1
    return A

test_impl()
```
locally on mainline and it doesn't seem to trip it?

Comment (DrTodd13) [reply numba/tests/test_parfors.py:2371]: This was something else that I thought should be tested even if it wasn't triggered by the original error.  I should remove the issue number from that test.

Comment (stuartarchibald) [reply numba/tests/test_parfors.py]: ah ok, it worked on mainline and it also worked with this patch, but reduction in nested parfors seems like a good thing to have a test for, happy to leave in with the issue number removed, thanks.

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch and fixes.

**Pull request merged**